src/PoolManager.sol

```diff
@@ -407,11 +385,9 @@ contract PoolManager is Auth, IPoolManager {
             undeployedTranche.decimals,
             trancheTokenWards
         );
-        address restrictionManager = restrictionManagerFactory.newRestrictionManager(
-            undeployedTranche.restrictionSet, token, restrictionManagerWards
-        );
-
-        TrancheTokenLike(token).file("restrictionManager", restrictionManager);
+        address restrictionManager =
+            restrictionManagerFactory.newRestrictionManager(undeployedTranche.hook, token, restrictionManagerWards);
+        TrancheTokenLike(token).file("hook", restrictionManager);
```

Comment:
from my understanding, we will add the initial restrictionManager deployment to the deploy script and simply pass to all tranche tokens

Reply:
What was  rather my idea is that we add an `address` arg to the `AddTranche` message, replacing the `restrictionSet`. So we always pass the restriction manager address here